packages/create-react-native-library/templates/native-common/android/build.gradle

```diff
@@ -19,41 +19,81 @@ buildscript {
 }
 
 def isNewArchitectureEnabled() {
-  return rootProject.hasProperty("newArchEnabled") &&  rootProject.getProperty("newArchEnabled") == "true"
+  return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
 }
 
-apply plugin: 'com.android.library'
+def resolveReactNativeDirectory() {
+  def reactNativeLocation = rootProject.hasProperty("reactNativeDir") ? rootProject.getProperty("reactNativeDir") : null
+
+  if (reactNativeLocation != null) {
+    return file(reactNativeLocation)
+  }
+
+  try {
+    // Resolve React Native location with Node
+    // This will make sure that we get installation location correctly in monorepos
+    def reactNativePackageJsonPathStdout = new ByteArrayOutputStream()
+
+    exec {
+      commandLine("node", "-p", "require.resolve('react-native/package.json')")
+      ignoreExitValue true
+      standardOutput = reactNativePackageJsonPathStdout
+    }
```

Comment:
Q: Why is this necessary now? How was this working on 0.70?
I would refrain from doing `exec{}` here as they're slowing down every build, for everyone and they get called for every library, so this gets exponentially worse.

Reply:
> we check the react native version to specify the correct dependency

How were you checking the rn version before .71?

Reply:
@cortinico previously we didn't need to do that. we check it now since some of the code is different based on the version. Here is a snippet from generated gradle config:

```gradle

def isNewArchitectureEnabled() {
  return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
}

def resolveReactNativeDirectory() {
  // ...
}

apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

if (isNewArchitectureEnabled()) {
  apply plugin: 'com.facebook.react'
}

def getExtOrDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : project.properties['ReactNativeTest_' + name]
}

def getExtOrIntegerDefault(name) {
  return rootProject.ext.has(name) ? rootProject.ext.get(name) : (project.properties['ReactNativeTest_' + name]).toInteger()
}

def reactNativeRootDir = resolveReactNativeDirectory()
def reactProperties = new Properties()
file("$reactNativeRootDir/ReactAndroid/gradle.properties").withInputStream { reactProperties.load(it) }

def REACT_NATIVE_VERSION = reactProperties.getProperty("VERSION_NAME")
def REACT_NATIVE_MAJOR_VERSION = REACT_NATIVE_VERSION.split("\\.")[0].toInteger()
def REACT_NATIVE_MINOR_VERSION = REACT_NATIVE_VERSION.split("\\.")[1].toInteger()

android {
  compileSdkVersion getExtOrIntegerDefault('compileSdkVersion')

  defaultConfig {
    minSdkVersion getExtOrIntegerDefault('minSdkVersion')
    targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
    buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
  }
  buildTypes {
    release {
      minifyEnabled false
    }
  }

  lintOptions {
    disable 'GradleCompatible'
  }

  compileOptions {
    sourceCompatibility JavaVersion.VERSION_1_8
    targetCompatibility JavaVersion.VERSION_1_8
  }

  sourceSets {
    main {
      if (isNewArchitectureEnabled()) {
          java.srcDirs += [
            // This is needed to build Kotlin project with NewArch enabled
            "${project.buildDir}/generated/source/codegen/java"
          ]
      }
    }
  }
}

repositories {
  mavenCentral()
  google()

  if (REACT_NATIVE_MAJOR_VERSION == 0 && REACT_NATIVE_MINOR_VERSION < 71) {
    def androidSourcesDir = null
    def androidSourcesName = 'React Native sources'

    if (rootProject.ext.has('reactNativeAndroidRoot')) {
      androidSourcesDir = rootProject.ext.get('reactNativeAndroidRoot')
    } else {
      androidSourcesDir = new File(reactNativeRootDir, '/android')
    }

    if (androidSourcesDir.exists()) {
      maven {
        url androidSourcesDir.toString()
        name androidSourcesName
      }

      logger.info("${project.name}:reactNativeAndroidRoot ${defaultDir.canonicalPath}")
    }
  }
}

def kotlin_version = getExtOrDefault('kotlinVersion')

dependencies {
  if (REACT_NATIVE_MAJOR_VERSION == 0 && REACT_NATIVE_MINOR_VERSION < 71) {
    //noinspection GradleDynamicVersion
    implementation 'com.facebook.react:react-native:+'
    // From node_modules
  } else {
    implementation 'com.facebook.react:react-android:' + REACT_NATIVE_VERSION
  }

  implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
}

if (isNewArchitectureEnabled()) {
  react {
    jsRootDir = file("../src/")
    libraryName = "ReactNativeTestView"
    codegenJavaPackageName = "com.bob.reactnativetest"
  }
}
```

Reply:
Thanks for clarifying. Yup that's a problem and we should look into it. Ideally the React Native Gradle Plugin should expose this information and you should be able to just consume it.

Reply:
@cortinico Gotcha! I'll keep an eye out and help with testing it

Reply:
So back on this, I think this whole logic can be cleaned to don't need `REACT_NATIVE_MINOR_VERSION` at all.

1. The `implementation` dependency can be just `implementation 'com.facebook.react:react-native:+'`. On 71+ the Gradle Plugin will take care of handling this.
2. The repositories block should be just "safe" in the sense that if the `/android` folder can't be found => No extra `maven{}` block should be called (which is already the case).

Reply:
I'll keep this in mind for the future and for .72 we might be able to expose `reactNativeDir` and `reactNativeVersion` to libraries via the Gradle Plugin